clang/test/CodeGen/PowerPC/ppc32-complex-soft-float-gnu-abi.c

```diff
@@ -0,0 +1,286 @@
+// CHECK-GNU-SOFT-FLOAT-NEXT:    [[CF_IMAG:%.*]] = load float, ptr [[CF_IMAGP]], align 4
+// CHECK-GNU-SOFT-FLOAT-NEXT:    [[CD_REALP:%.*]] = getelementptr inbounds { double, double }, ptr [[CD]], i32 0, i32 0
+// CHECK-GNU-SOFT-FLOAT-NEXT:    [[CD_REAL:%.*]] = load double, ptr [[CD_REALP]], align 8
+// CHECK-GNU-SOFT-FLOAT-NEXT:    [[CD_IMAGP:%.*]] = getelementptr inbounds { double, double }, ptr [[CD]], i32 0, i32 1
+// CHECK-GNU-SOFT-FLOAT-NEXT:    [[CD_IMAG:%.*]] = load double, ptr [[CD_IMAGP]], align 8
+// CHECK-GNU-SOFT-FLOAT-NEXT:    [[COERCE_REALP:%.*]] = getelementptr inbounds { float, float }, ptr [[COERCE]], i32 0, i32 0
+// CHECK-GNU-SOFT-FLOAT-NEXT:    [[COERCE_IMAGP:%.*]] = getelementptr inbounds { float, float }, ptr [[COERCE]], i32 0, i32 1
+// CHECK-GNU-SOFT-FLOAT-NEXT:    store float [[CF_REAL]], ptr [[COERCE_REALP]], align 4
+// CHECK-GNU-SOFT-FLOAT-NEXT:    store float [[CF_IMAG]], ptr [[COERCE_IMAGP]], align 4
+// CHECK-GNU-SOFT-FLOAT-NEXT:    [[TMP1:%.*]] = load [1 x i64], ptr [[COERCE]], align 4
+// CHECK-GNU-SOFT-FLOAT-NEXT:    [[COERCE1_REALP:%.*]] = getelementptr inbounds { double, double }, ptr [[COERCE1]], i32 0, i32 0
+// CHECK-GNU-SOFT-FLOAT-NEXT:    [[COERCE1_IMAGP:%.*]] = getelementptr inbounds { double, double }, ptr [[COERCE1]], i32 0, i32 1
+// CHECK-GNU-SOFT-FLOAT-NEXT:    store double [[CD_REAL]], ptr [[COERCE1_REALP]], align 8
+// CHECK-GNU-SOFT-FLOAT-NEXT:    store double [[CD_IMAG]], ptr [[COERCE1_IMAGP]], align 8
+// CHECK-GNU-SOFT-FLOAT-NEXT:    [[TMP2:%.*]] = load [4 x i32], ptr [[COERCE1]], align 8
+// CHECK-GNU-SOFT-FLOAT-NEXT:    [[CALL:%.*]] = call [4 x i32] @testComplexDouble(float noundef [[TMP0]], [1 x i64] noundef [[TMP1]], [4 x i32] noundef [[TMP2]])
+// CHECK-GNU-SOFT-FLOAT-NEXT:    store [4 x i32] [[CALL]], ptr [[COERCE2]], align 8
+// CHECK-GNU-SOFT-FLOAT-NEXT:    [[COERCE2_REALP:%.*]] = getelementptr inbounds { double, double }, ptr [[COERCE2]], i32 0, i32 0
+// CHECK-GNU-SOFT-FLOAT-NEXT:    [[COERCE2_REAL:%.*]] = load double, ptr [[COERCE2_REALP]], align 8
+// CHECK-GNU-SOFT-FLOAT-NEXT:    [[COERCE2_IMAGP:%.*]] = getelementptr inbounds { double, double }, ptr [[COERCE2]], i32 0, i32 1
+// CHECK-GNU-SOFT-FLOAT-NEXT:    [[COERCE2_IMAG:%.*]] = load double, ptr [[COERCE2_IMAGP]], align 8
+// CHECK-GNU-SOFT-FLOAT-NEXT:    [[RETVAL_REALP:%.*]] = getelementptr inbounds { double, double }, ptr [[RETVAL]], i32 0, i32 0
+// CHECK-GNU-SOFT-FLOAT-NEXT:    [[RETVAL_IMAGP:%.*]] = getelementptr inbounds { double, double }, ptr [[RETVAL]], i32 0, i32 1
+// CHECK-GNU-SOFT-FLOAT-NEXT:    store double [[COERCE2_REAL]], ptr [[RETVAL_REALP]], align 8
+// CHECK-GNU-SOFT-FLOAT-NEXT:    store double [[COERCE2_IMAG]], ptr [[RETVAL_IMAGP]], align 8
+// CHECK-GNU-SOFT-FLOAT-NEXT:    [[TMP3:%.*]] = load [4 x i32], ptr [[RETVAL]], align 8
+// CHECK-GNU-SOFT-FLOAT-NEXT:    ret [4 x i32] [[TMP3]]
+//
+_Complex double checkComplexDoubleOnStack(float x1, _Complex float cf, float x2, _Complex double cd)
```

Comment:
I think we only the test senario is enough to test the `float` parameter occups one gpr in the `PPC32_SVR4_ABIInfo::classifyArgumentType`. We can delete all other scenarios.